pkg/agent/testdata/AKSUbuntu1604+CustomKubeletConfig+CustomLinuxOSConfig/line40.sh

```diff
@@ -560,7 +560,21 @@ installKubeletKubectlAndKubeProxy() {
         if [[ "$install_default_if_missing" == true ]]; then
             if [[ ! -z ${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER} ]]; then
                 echo "Detect Bootstrap profile artifact is Cache, will use oras to pull artifact binary"
-                registry_url="${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER}/${K8S_REGISTRY_REPO}/kubernetes-node:v${KUBERNETES_VERSION}-linux-${CPU_ARCH}"
+                binary_version="v${KUBERNETES_VERSION}" 
+                url_regax='https://acs-mirror.azureedge.net/kubernetes/v[0-9]+\.[0-9]+\..+/binaries/.+'
+                if [[ -n ${KUBE_BINARY_URL} ]]; then
+                    if [[ ${KUBE_BINARY_URL} =~ $registry_regex ]]; then
+                        version_with_prefix="${KUBE_BINARY_URL#*kubernetes/}"
+                        binary_version="${version_with_prefix%%/*}"
+                        echo "Extracted version: $binary_version from KUBE_BINARY_URL: ${KUBE_BINARY_URL}"
+                    fi
+                fi
+                registry_url="${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER}/${K8S_REGISTRY_REPO}/kubernetes-node:${binary_version}-linux-${CPU_ARCH}"
+
+                if [[ -n "${KUBE_BINARY_URL}" ]] && isRegistryUrl "${KUBE_BINARY_URL}"; then
```

Comment:
why we need this change?

Reply:
do you mean `if [[ -n "${KUBE_BINARY_URL}" ]] && isRegistryUrl "${KUBE_BINARY_URL}"; then`?
It is for possible in the future rp directly pass registry url instead of acs-mirror if it is ni cluster